src/wallet/wallet2.cpp

```diff
@@ -8583,6 +8583,52 @@ bool wallet2::get_tx_key(const crypto::hash &txid, crypto::secret_key &tx_key, s
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool wallet2::set_tx_key(const crypto::hash &txid, const crypto::secret_key &tx_key, const std::vector<crypto::secret_key> &additional_tx_keys)
+{
+  THROW_WALLET_EXCEPTION_IF(m_tx_keys.find(txid) != m_tx_keys.end(), error::wallet_internal_error, "Tx key for is already stored in the cache");
```

Comment:
That means if you make a mistake, you can't correct it.

Reply:
ok